### PR TITLE
Feat: add logrotate conf for mgr-create-bootstrap-repo

### PR DESCRIPTION
**Pull request by mbologna**
## What does this PR change?

* make mgr-create-bootstrap-repo log into a new separate directory /var/log/rhn/mgr-create-bootstrap-repo/ (for reviewers: the directory is created by `rhnLog.initLOG`)
* `log`-rotate those files by day.

NOTE: one minor problem: existing users that upgrade the package will not have the already existing files in /var/log/rhn/mgr-create-bootstrap-repo.* cleaned up.
Reviewers: is cleaning up those files needed? (Maybe in the specfile)

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: https://github.com/uyuni-project/uyuni-docs/blob/master/modules/reference/pages/command-line-tools.adoc does not mention log location
- [X] **DONE**

## Test coverage
- No tests: already covered by Cucumber

- [X] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/12990

- [x] **DONE**

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: susemanager/etc/logrotate.d/susemanager-tools [hunk 1]
```diff
@@ -11,4 +11,14 @@
...
-
+/var/log/rhn/mgr-create-bootstrap-repo/mgr-create-bootstrap-repo.log {
+    compress
+    dateext
+    daily
+    maxage 30
+    missingok
+    notifempty
+    rotate 30
+    size 4096k
```
Comment (mcalmer) [inline susemanager/etc/logrotate.d/susemanager-tools:22]: I think size could make problems.
We want to rotate daily? So no need to specify a size limit.

File: susemanager/etc/logrotate.d/susemanager-tools [hunk 2]
```diff
@@ -11,4 +11,14 @@
...
+    su wwwrun www
```
Comment (mcalmer) [inline susemanager/etc/logrotate.d/susemanager-tools:23]: I think we do not need the "su" option. I think it is more important to give the new directory the correct permissions.
Under which user is mgr-create-bootstrap-repo normally executed?

AFAIK reposync is called via taskomatic which should use root/root. When run manually on the commandline, also root should do it.
No `su` and directory owned by `root` should be fine.
Needs to be tested.

File: susemanager/etc/logrotate.d/susemanager-tools (hunk 1 above)
Comment (mbologna) [reply susemanager/etc/logrotate.d/susemanager-tools:22]: Agreed, I will remove it.

File: susemanager/etc/logrotate.d/susemanager-tools (hunk 2 above)
Comment (mbologna) [reply susemanager/etc/logrotate.d/susemanager-tools:23]: More data:

`mgr-create-bootstrap-repo` is called interactively by `root` (assuming the user is using that user to call it) and by reposync https://github.com/uyuni-project/uyuni/blob/master/backend/satellite_tools/reposync.py#L681

reposync, when scheduled from the WebUI, is running as `root`:

```
root     27529  9423 99 20:03 ?        00:03:26 /usr/bin/python3 -u /usr/bin/spacewalk-repo-sync --channel sle-module-basesystem15-sp3-pool-x86_64-sms-4.2 --type yum --non-interactive
```

This means that the directory storing the logs `/var/log/rhn/mgr-create-bootstrap-repo/` is created with `uid=wwwrun` (or `uid=apache` for non-SUSE) and `gid=0` https://github.com/uyuni-project/uyuni/blob/master/backend/common/rhnLog.py#L109-L110:

```
# ls -dl /var/log/mgr-create-bootstrap-repo
drwxr-xr-x 2 wwwrun root 43 Mar  3 20:10 mgr-create-bootstrap-repo
# ls -l /var/log/mgr-create-bootstrap-repo/mgr-create-bootstrap-repo
total 12
-rw-rw---- 1 wwwrun root 9744 Mar  3 20:11 mgr-create-bootstrap-repo.log
```

Test without 'su' option:

```
# date -s 'tomorrow'
# date
Thu  4 Mar 20:15:10 CET 2021
# logrotate /etc/logrotate.conf
# ls -l /var/log/rhn/mgr-create-bootstrap-repo
-rw-rw---- 1 wwwrun root    0 Mar  4 20:15 mgr-create-bootstrap-repo.log
-rw-rw---- 1 wwwrun root 1460 Mar  3 20:14 mgr-create-bootstrap-repo.log-20210304.xz
```

**Comment by mcalmer** [comment]
Hmm, this is bad. Than we need to keep the su option.

**Review by mcalmer — CHANGES_REQUESTED** [review]
Seems we need to keep the su option

**Comment by mbologna** [comment]
TODO: add text about old logs in release-notes

**Review by mcalmer — APPROVED** [review]

**Comment by mbologna** [comment]
> TODO: add text about old logs in release-notes

https://gitlab.suse.de/documentation/release-notes-suse-manager/-/merge_requests/55/diffs

**Pull request merged**